src/test/java/com/ncc/aif/ExamplesAndValidationTest.java

```diff
@@ -1821,8 +1824,65 @@ void validEventRelationAndEventRelationEdge() {
                 markImportance(relationEdge, 102);
 
                 makeValidHypothesis(entity, event, eventEdge, relation, relationEdge);
-                testValid("NISTHypothesis.valid: Each hypothesis graph must have at least one " +
-                        "event or relation with at least one edge.");
+                testValid("NISTHypothesis.valid (event has event edge and relation has relation edge): Each " +
+                        "hypothesis graph must have at least one event or relation with at least one edge.");
+            }
+        }
+
+        // Clusters must be homogeneous by base class (Entity, Event, or Relation)
+        @Nested
+        class HypothesisClustersMustBeHomogeneous {
+
+            @Test
+            void invalid() {
+
+                final Resource relation = makeRelation(model, getUri("relation-1"), system);
+                final Resource relationCluster = makeClusterWithPrototype(model, getClusterUri(), relation,
+                        "Relation", system);
+
+                markImportance(relationCluster, 103);
+                markJustification(addType(relation, SeedlingOntology.GeneralAffiliation_APORA), justification);
+
+                final Resource relationEdge = markAsArgument(model, relation, SeedlingOntology.GeneralAffiliation_APORA_Affiliate,
+                        entity, system, 1d, getAssertionUri());
+                markImportance(relationEdge, 102);
```

Comment:
Everything from here, up is duplicated in valid(). This should be moved into a setup() function to only have the code live in one place.

Reply:
Updated.